src/Compilers/CSharp/Portable/FlowAnalysis/UnassignedFieldsWalker.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed class UnassignedFieldsWalker : DataFlowPass
+    {
+        private readonly DiagnosticBag _diagnostics;
+
+        private UnassignedFieldsWalker(CSharpCompilation compilation, MethodSymbol method, BoundNode node, DiagnosticBag diagnostics)
+            : base(compilation, method, node, trackClasses: true)
```

Comment:
>trackClasses [](start = 46, length = 12)

I am not sure what this name is supposed to convey. Are we not passing true for structures?

Reply:
Renamed `_trackClassFields` and removed the confusing uses we discussed.